File: gym-management-api/src/main/java/com/gym/api/gym_management_api/domain/model/Academia.java
package com.gym.api.gym_management_api.domain.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Academia {
    
  private Long id;
  private String nome;
  private String endereco;
  private String telefone;
  private Long adminId;
  private Boolean ativa;
  private LocalDateTime dataCadastro;
  private LocalDateTime dataAtualizacao;
  
  public void ativar() {
    this.ativa = true;
  }
  
  public void inativar() {
    this.ativa = false;
  }
  
}

File: client/src/types/academia.ts
export interface AcademiaResponse {
  id: number;
  nome: string;
  endereco: string | null;
  telefone: string | null;
  adminId: number;
  ativa: boolean;
  dataCadastro: string;
}

export interface CreateAcademiaRequest {
  nome: string;
  endereco?: string;
  telefone?: string;
}
